DEVELOPERS.md

```diff
@@ -0,0 +1,175 @@
+![Circus Train BigQuery.](circus-train-bigquery.png "Moving data from Google BigQuery to Hive.")
+
+## Overview 
+Circus Train BigQuery is an extension of Circus Train which allows tables in Google BigQuery to be replicated.
+
+This file contains a collection of notes put together by developers who have worked on BigQuery to provide some explanation to the code. This file is not completely exhaustive of all the inner workings of the project, so do feel free to add more information or detail. 
+
+
+## README.md
+
+First and foremost, its worth having a read through the Circus Train [README.md](https://github.com/HotelsDotCom/circus-train) file and the Circus Train BigQuery [README.md](https://github.com/HotelsDotCom/circus-train-bigquery/blob/master/README.md). These are pretty extensive guides containing a lot of info on the projects, including how to run each of them and the different configurations which can be used. 
+
+It may also be useful to read through the DEVELOPERS.md file for Circus Train. 
+
+
+## Basic description
+Circus Train BigQuery handles applying any partition filters to the data in the Google BigQuery table and extracts it as Avro into Google Cloud Storage, (GCS). Circus Train listeners are used to convert the Google BigQuery metadata into a Hive table object. The data is then replicated from source to target using the metadata from this mocked Hive table.
```

Comment:
I think we first need a very high level description, then flesh it out with some of the additional info you have here. Something like "At a high level CTBQ uses X integration points provided by CT (mention them, the copiers or whatever) to replicate data out of BigQuery and into Hive. It first runs a query to extract the data out of BQ into a temporary BQ table (applying any necessary partition filters) and then uses the BQ X API (find out which one the extractor uses) to extract that data out of this temporary table onto GCS as Avro files. A Hive table _object_ (i.e. not a real Hive table) is then put "on top" of this data and Circus train is instructed to replicate the data from this source to the target as normal."